paddle/fluid/operators/fetch_barrier_op.cc

```diff
@@ -0,0 +1,107 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <future>  // NOLINT
+#include <ostream>
+
+#include "paddle/fluid/framework/data_type.h"
```

Comment:
I think we might only need `send` and `recv` ops and adding attrs to control's it's behavior so that we won't copy these codes around.

Reply:
Good idea, but it's hard for the current implement of SSAGraph, it's use op type and input/output argument to determine the dependency, we need to do more things to merge `send_op`, `send_vars_op`, `batch_barrier`, `recv_op`, `fetch_barrier`. I created an issue to follow this comment.